### DISCUSSION TITLE
Live MIDI transpose from midi input or chord track

### DISCUSSION THREAD

**Discussion by viessg** [SynthstromAudible/DelugeFirmware#2699] [Ideas]
Hello, I'm new to Deluge firmware community - so I don't know, maybe it was discussed already - I just found live midi FX topic somewhat similar.
Could it be possible to transpose particular midi tracks (maybe there could be two options Midi listen as it is now, and Midi transpose) while sticking pre-recorded midi content within scale - according to chords (harmonic content) received from external midi source or dedicated internal chord track? Somewhat arranger synths feature? Was such topic discussed here - or this can go under Live midi FX section?

**Comment by m-m-adams** [comment]
If you set a midi track to TRANSPOSE I think it does what you want? It transposes the song according to notes it receives

**Reply by m-m-adams** [reply]
No following chords yet though

**Comment by duncanrmi** [comment]
gonna stick this here for the moment, see if anyone else is actually using this function-

internal transpose. I've got a couple of nice arpeggiated acoustic guitar parts with the deluge doing the arpeggiation, based on note clusters with probabilities set so that it gets a different choice of notes to arpeggiate each pass, & some randomness chucked in with the new arpeg features. so far so good. & a bass synth line.

so now I'm trying the internal transpose & have a couple of observations-
1) if you go back into any of the clips being transposed & try to edit them.... impossible. I know they're being transposed, but can we just leave the notes where they were in the untransposed version instead of having them leaping around as they get transposed?
2) the 'notes' doing the transposing seem to need to be a few beats early or they don't get 'processed' & nothing transposes. 
3) the 'notes' doing the transposing seem to be issuing note-on/offs that are affecting the clips being transposed, so that long release notes are getting chopped off.

this is in the latest nightly, chopin.
